src/MPQEditor/MPQCircleSelector.ts

```diff
@@ -0,0 +1,320 @@
+      panel,
+      extensionUri,
+      args,
+      handler
+    );
+
+    MPQSelectionPanel.panels.push(graphSelPanel);
+    graphSelPanel.loadContent();
+    graphSelPanel.onForwardSelection(args.names);
+  }
+
+  private static findSelPanel(
+    docPath: string,
+    id: string
+  ): MPQSelectionPanel | undefined {
+    let result = undefined;
+    MPQSelectionPanel.panels.forEach((selpan) => {
+      if (docPath === selpan._modelPath && id === selpan._ownerId) {
+        result = selpan;
+      }
+    });
+    return result;
+  }
+
+  /**
+   * @brief called when owner is closing
+   */
+  public static closeByOwner(
+    extensionUri: vscode.Uri,
+    args: MPQSelectionCmdCloseArgs
+  ) {
+    let selPanel = MPQSelectionPanel.findSelPanel(
+      args.modelPath,
+      args.document.uri.toString()
+    );
+    if (selPanel) {
+      selPanel.dispose();
+    }
+  }
+
+  /**
+   * @brief called when owner loads visq file
+   */
+  public static loadVisqInfo(
+    extensionUri: vscode.Uri,
+    args: MPQSelectionCmdVisqArgs
+  ) {
+    let selPanel = MPQSelectionPanel.findSelPanel(
+      args.modelPath,
+      args.document.uri.toString()
+    );
+    if (selPanel) {
+      selPanel.onLoadVisqInfo(args.visqPath);
+    }
+  }
+
+  /**
+   * @brief called when owner selection state of nodes has changed
+   */
+  public static forwardSelectionByOwner(
+    extensionUri: vscode.Uri,
+    args: MPQSelectionCmdLayersChangedArgs
+  ) {
+    let selPanel = MPQSelectionPanel.findSelPanel(
+      args.modelPath,
+      args.document.uri.toString()
+    );
+    if (selPanel) {
+      selPanel.onForwardSelection(args.names);
+    }
+  }
+
+  private constructor(
+    panel: vscode.WebviewPanel,
+    extensionUri: vscode.Uri,
+    args: MPQSelectionCmdOpenArgs,
+    handler: MPQSelectionEvent | undefined
+  ) {
+    super(extensionUri, panel.webview);
+
+    // let parsedPath = path.parse(args.docPath);
+    // let fileBase = path.join(parsedPath.dir, parsedPath.name);
+
+    this._panel = panel;
+    this._ownerId = args.document.uri.toString();
+    this._document = args.document;
+    this._modelPath = args.modelPath;
+    this._mpqEventHandler = handler;
+    this._lastSelected = args.names;
+
+    // Listen for when the panel is disposed
+    // This happens when the user closes the panel or when the panel is closed programmatically
+    this._panel.onDidDispose(() => this.dispose(), null, this._disposables);
+
+    // Update the content based on view changes
+    this._panel.onDidChangeViewState(
+      () => {
+        if (this._panel.visible) {
+          // NOTE if we call this.update(), it'll reload the model which may take time.
+          // TODO call conditional this.update() when necessary.
+          // this.update();
+        }
+      },
+      null,
+      this._disposables
+    );
+
+    this.initGraphCtrl(this._modelPath, this);
+    this.setMode("selector");
+    //this.setMode("visqselr");
```

Comment:
?

Reply:
It was an experiment to set new mode at once on html creation, but i failed so i'll remove it.